experimental/bridge/opentracing/lib.go

```diff
@@ -0,0 +1,399 @@
+	e.pairs = append(e.pairs, otTagToOtelCoreKeyValue(key, value))
+}
+
+func otLogFieldsToOtelCoreKeyValues(fields []otlog.Field) []otelcore.KeyValue {
+	encoder := &bridgeFieldEncoder{}
+	for _, field := range fields {
+		field.Marshal(encoder)
+	}
+	return encoder.pairs
+}
+
+func (s *bridgeSpan) LogKV(alternatingKeyValues ...interface{}) {
+	fields, err := otlog.InterleavedKVToFields(alternatingKeyValues...)
+	if err != nil {
+		return
+	}
+	s.LogFields(fields...)
+}
+
+func (s *bridgeSpan) SetBaggageItem(restrictedKey, value string) ot.Span {
+	s.ctx.setBaggageItem(restrictedKey, value)
+	return s
+}
+
+func (s *bridgeSpan) BaggageItem(restrictedKey string) string {
+	return s.ctx.baggageItem(restrictedKey)
+}
+
+func (s *bridgeSpan) Tracer() ot.Tracer {
+	return s.tracer
+}
+
+func (s *bridgeSpan) LogEvent(event string) {
+	s.LogEventWithPayload(event, nil)
+}
+
+func (s *bridgeSpan) LogEventWithPayload(event string, payload interface{}) {
+	data := ot.LogData{
+		Event:   event,
+		Payload: payload,
+	}
+	s.Log(data)
+}
+
+func (s *bridgeSpan) Log(data ot.LogData) {
+	record := data.ToLogRecord()
+	s.LogFields(record.Fields...)
+}
+
+type bridgeTracer struct {
+	otelTracer oteltrace.Tracer
+}
+
+var _ ot.Tracer = &bridgeTracer{}
+
+func NewTracer(tracer oteltrace.Tracer) ot.Tracer {
+	return &bridgeTracer{
+		otelTracer: tracer,
+	}
+}
+
+func NewTracerFromGlobal() ot.Tracer {
+	return NewTracer(oteltrace.GlobalTracer())
+}
+
+func (t *bridgeTracer) StartSpan(operationName string, opts ...ot.StartSpanOption) ot.Span {
+	sso := ot.StartSpanOptions{}
+	for _, opt := range opts {
+		opt.Apply(&sso)
+	}
+	// TODO: handle links, needs SpanData to be in the API first?
+	reference, _ := otSpanReferencesToOtelReferenceAndLinks(sso.References)
+	// TODO: handle span kind, needs SpanData to be in the API first?
+	attributes, _, hadTrueErrorTag := otTagsToOtelAttributesKindAndError(sso.Tags)
+	// TODO: Eurgh, in opentracing you create the span first and
+	// then create a context out of it. In opentelemetry, you
+	// create a context first, then pass it to the Span
+	// constructor. Also, what to do then with the returned context?
+	_, otelSpan := t.otelTracer.Start(context.TODO(), operationName, func(opts *oteltrace.SpanOptions) {
+		opts.Attributes = attributes
+		opts.StartTime = sso.StartTime
+		opts.Reference = reference
+		opts.RecordEvent = true
+	})
+	if hadTrueErrorTag {
+		otelSpan.SetStatus(codes.Unknown)
+	}
+	return &bridgeSpan{
+		otelSpan: otelSpan,
+		ctx: &bridgeSpanContext{
+			otelSpanContext: otelSpan.SpanContext(),
+		},
+		tracer: t,
+	}
+}
+
+type spanKindTODO struct{}
+
+func otTagsToOtelAttributesKindAndError(tags map[string]interface{}) ([]otelcore.KeyValue, spanKindTODO, bool) {
+	kind := spanKindTODO{}
+	error := false
+	var pairs []otelcore.KeyValue
+	for k, v := range tags {
+		switch k {
+		case string(otext.SpanKind):
+			// TODO: java has some notion of span kind, it
```

Comment:
This level of semantic convention hasn't been spec'd very well, I'd wait for more discussion.